ext/BijectorsEnzymeCoreExt.jl

```diff
@@ -0,0 +1,184 @@
+module BijectorsEnzymeCoreExt
+
+if isdefined(Base, :get_extension)
+    using EnzymeCore:
+        Active,
+        Const,
+        Duplicated,
+        DuplicatedNoNeed,
+        BatchDuplicated,
+        BatchDuplicatedNoNeed,
+        EnzymeRules
+    using Bijectors: find_alpha
+else
+    using ..EnzymeCore:
+        Active,
+        Const,
+        Duplicated,
+        DuplicatedNoNeed,
+        BatchDuplicated,
+        BatchDuplicatedNoNeed,
+        EnzymeRules
+    using ..Bijectors: find_alpha
+end
+
+# Compute a tuple of partial derivatives wrt non-`Const` arguments
+# and `nothing`s for `Const` arguments
+function ∂find_alpha(
+    Ω::Real,
+    wt_y::Union{Const,Active,Duplicated,BatchDuplicated},
+    wt_u_hat::Union{Const,Active,Duplicated,BatchDuplicated},
+    b::Union{Const,Active,Duplicated,BatchDuplicated},
+)
+    # We reuse the following term in the computation of the derivatives
+    Ωpb = Ω + b.val
+    c = wt_u_hat.val * sech(Ωpb)^2
+    cp1 = c + 1
+
+    ∂Ω_∂wt_y = wt_y isa Const ? nothing : oneunit(wt_y.val) / cp1
+    ∂Ω_∂wt_u_hat = wt_u_hat isa Const ? nothing : -tanh(Ωpb) / cp1
+    ∂Ω_∂b = b isa Const ? nothing : -c / cp1
+
+    return (∂Ω_∂wt_y, ∂Ω_∂wt_u_hat, ∂Ω_∂b)
+end
+
+# `muladd` for partial derivatives that can deal with `nothing` derivatives
+_muladd_partial(::Nothing, ::Const, x::Union{Real,Tuple{Vararg{Real}},Nothing}) = x
+_muladd_partial(x::Real, y::Duplicated, z::Real) = muladd(x, y.dval, z)
+_muladd_partial(x::Real, y::Duplicated, ::Nothing) = x * y.dval
+function _muladd_partial(x::Real, y::BatchDuplicated{<:Real,N}, z::NTuple{N,Real}) where {N}
+    let x = x
+        map((a, b) -> muladd(x, a, b), y.dval, z)
+    end
+end
+_muladd_partial(x::Real, y::BatchDuplicated, ::Nothing) = map(Base.Fix1(*, x), y.dval)
+
+function EnzymeRules.forward(
+    ::Const{typeof(find_alpha)},
+    ::Type{RT},
+    wt_y::Union{Const,Duplicated,BatchDuplicated},
+    wt_u_hat::Union{Const,Duplicated,BatchDuplicated},
+    b::Union{Const,Duplicated,BatchDuplicated},
+) where {RT<:Union{Const,Duplicated,DuplicatedNoNeed,BatchDuplicated,BatchDuplicatedNoNeed}}
+    # Check that the types of the activities are consistent
+    if !(
+        RT <: Union{Const,Duplicated,DuplicatedNoNeed} &&
+        wt_y isa Union{Const,Duplicated} &&
+        wt_u_hat isa Union{Const,Duplicated} &&
+        b isa Union{Const,Duplicated}
+    ) && !(
+        RT <: Union{Const,BatchDuplicated,BatchDuplicatedNoNeed} &&
+        wt_y isa Union{Const,BatchDuplicated} &&
+        wt_u_hat isa Union{Const,BatchDuplicated} &&
+        b isa Union{Const,BatchDuplicated}
+    )
+        throw(ArgumentError("inconsistent activities"))
+    end
+
+    # Compute primal value
+    Ω = find_alpha(wt_y.val, wt_u_hat.val, b.val)
+
+    # Early exit if no derivatives are requested
+    if RT <: Const
+        return Ω
+    end
+
+    Ω̇ = if wt_y isa Const && wt_u_hat isa Const && b isa Const
```

Comment:
with the bump to 0.13 you can probably do !needs_shadow || (the const conditions)